### PR TITLE
Dpev/geozone c interface

### PR DESCRIPTION
**Pull request by dpev**
udMath given c-compatible declarations for c compilers
Migration of GeoZone to C interface (with c++ inlines for deprecated functions)

### CODE REVIEW

File: Include/udPlatform.h [hunk 1]
```diff
@@ -47,6 +47,7 @@
 #  error "PTHREADS are not being used!"
 # endif
 #elif defined(_MSC_VER) || defined(__MINGW32__)
+#define NOMINMAX
```
Comment (jproust) [inline Include/udPlatform.h:50]: where is this used ?

File: Source/udJSON.cpp [hunk 2]
```diff
@@ -186,94 +186,101 @@ udResult udJSON::SetObject()
...
-      break;
-  }
-epilogue:
-  return result;
+  return udJSON_SetArrayHelper(this, pArray, length, shrinkMathTypes);
 }
 
+//// ****************************************************************************
+//// Author: Dave Pevreal, May 2017
+//udResult udJSON::Set(const udDouble4x4 &v, bool shrink)
```
Comment (jproust) [inline Source/udJSON.cpp:243]: any reasons to keep it ?

File: udTest/src/udFileTests.cpp [hunk 3]
```diff
@@ -38,7 +37,7 @@ TEST(udFileTests, GeneralFileTests)
   EXPECT_EQ(udR_Success, udFileExists(pFilename, &size, &modifyTime));
 
   EXPECT_EQ(18, size);
-  EXPECT_GT(2, udAbs(currentTime - modifyTime));
+  EXPECT_GT(2, currentTime - modifyTime);
```
Comment (jproust) [inline udTest/src/udFileTests.cpp:40]: can't this be negative ? i don't know just asking. Are `modifyTime` and `currentTime` local time ?

File: Include/udPlatform.h (hunk 1 above)
Comment (dpev) [reply Include/udPlatform.h:50]: This is to stop windows #defining min and max which causes all sorts of calamity, including preventing the use of std::min and std::max.

File: Source/udJSON.cpp (hunk 2 above)
Comment (dpev) [reply Source/udJSON.cpp:243]: Oh I didn't see this, thanks. No the JSON reader is to remove all references to udMath

File: udTest/src/udFileTests.cpp (hunk 3 above)
Comment (dpev) [reply udTest/src/udFileTests.cpp]: That's interesting. Looking at it a second time it should fail always, I'll look into it.

File: Source/udJSON.cpp (hunk 2 above)
Comment (dpev) [reply Source/udJSON.cpp]: Removed now

File: udTest/src/udFileTests.cpp (hunk 3 above)
Comment (dpev) [reply udTest/src/udFileTests.cpp]: I changed the test to simply ensure the modifyTime is >= to currentTime

**Review by samsinsane — APPROVED** [review]

**Pull request merged**